### PR TITLE
[SG-24591] Add in product CTAs to the Cloud app

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
### Description
- Add new components `MarketingBlock` and `SelfHostedCta`, use them to add in product CTAs to the Cloud app

### Refs
- https://github.com/sourcegraph/sourcegraph/issues/24591

### CODE REVIEW

File: client/web/src/components/MarketingBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import * as React from 'react'
+
+import styles from './MarketingBlock.module.scss'
+
+interface MarketingBlockProps {
+    wrapperClassName?: string
+    contentClassName?: string
+}
+
+export const MarketingBlock: React.FunctionComponent<MarketingBlockProps> = ({
```
Comment (valerybugakov) [inline client/web/src/components/MarketingBlock.tsx:11]: Let's follow the file structure pattern that we use in the Wildcard package — a folder for each component:

```
client/web/src/components/MarketingBlock/index.ts
client/web/src/components/MarketingBlock/MarketingBlock.tsx
client/web/src/components/MarketingBlock/MarketingBlock.story.tsx
client/web/src/components/MarketingBlock/MarketingBlock.module.scss
```

File: client/web/src/components/SelfHostedCta.story.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { storiesOf } from '@storybook/react'
+import React from 'react'
+
+import { SelfHostedCta } from './SelfHostedCta'
+import { WebStory } from './WebStory'
+
+const { add } = storiesOf('web/SelfHostedCta', module).addDecorator(story => (
```
Comment (valerybugakov) [inline client/web/src/components/SelfHostedCta.story.tsx:7]: Let's move marketing stories into their own Storybook folder: 

```suggestion
const { add } = storiesOf('web/markering/SelfHostedCta', module).addDecorator(story => (
```

**Review by valerybugakov — COMMENTED** [review]
The spec looks good. 
Left some minor comments about the file structure and Storybook stories API to use.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov Could we have access to buildkite to see what is going wrong with the PR? So we won't have to ask you every time like this

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, we understand that this approach doesn't scale, and [we're working on the solution](https://github.com/sourcegraph/sourcegraph/issues/16647). I hope that we will enable public access to our Buildkite logs or will be able to give you access to our organization soon.

**Comment by gitstart-sourcegraph** [comment]
Thanks @valerybugakov , got it. Then could you please share the issue from buildkite?

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, one thing that we would do for now — run CI checks locally. Right now prettier check is failing, which can be run locally by `yarn prettier-check`.

Here's the list of steps to run:

```sh
# to generate Typescript types
yarn generate 

# Verify Typescript build
yarn build-ts

# Check linters
yarn all:eslint
yarn prettier-check
yarn all:stylelint
yarn graphql-lint

# Run unit tests
yarn test

# Prepare web application for integration tests
ENTERPRISE=1 DISABLE_TYPECHECKING=true yarn run build-web
# Run integration tests
yarn test-integration
```

Let me know if you bump into issues.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov Thanks for the list of checks, we are setting up those checks on our side CI.

**Comment by valerybugakov** [comment]
@gitstart-sourcegraph, there're a couple of unexpected changes according to [the Chromatic diff](https://www.chromatic.com/pullrequest?appId=5f0f381c0e50750022dc6bf7&number=25265&view=changes). I've invited the Gitstart team to our organization to view these diffs. Please let me know if it's not working for you.

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov Updated PR to fix unexpected UI changes.

File: client/web/src/components/SelfHostedCta/SelfHostedCta.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                        href="https://docs.sourcegraph.com/admin/install"
+                        {...linkProps}
+                    >
+                        self-hosted installation
+                    </a>
+                    .
+                </span>
+            </div>
+        )
+    }
```
Comment (valerybugakov) [inline client/web/src/components/SelfHostedCta/SelfHostedCta.tsx:48]: It looks like a good candidate for a standalone component. Is it possible to extract it into a `<SelfHostedCtaLink />` located in `client/web/src/components/SelfHostedCtaLink`? Then `selfHostedInstallLinkOnly` will be redundant here.

**Review by valerybugakov — COMMENTED** [review]
Looks great! Thanks for all the fixes. 
Left a couple of minor comments. After them and the design review, it's good to go.

File: client/web/src/org/OrgsArea.tsx [hunk 4]
```diff
@@ -50,6 +51,11 @@ interface Props
 export const OrgsArea: React.FunctionComponent<Props> = props => (
     <Switch>
         <Route path={`${props.match.url}/new`} component={NewOrganizationPage} exact={true} />
+        <Route
+            path={`${props.match.url}/about`}
+            render={routeComponentProps => <AboutOrganizationPage {...props} {...routeComponentProps} />}
+            exact={true}
+        />
```
Comment (valerybugakov) [inline client/web/src/org/OrgsArea.tsx:58]: According to [mockups](https://www.figma.com/file/IgD7ZzmNt4Un2nqkk1u1X4/Self-hosted-CTAs-in-the-Cloud-product?node-id=214%3A657) we need to display the sidebar next to this page. Probably this route should be moved to `client/web/src/user/settings/routes.tsx`.

File: client/web/src/user/settings/UserSettingsSidebar.tsx [hunk 5]
```diff
@@ -83,23 +82,16 @@ export const UserSettingsSidebar: React.FunctionComponent<UserSettingsSidebarPro
...
-                                >
-                                    Learn More <ExternalLinkIcon className="icon-inline" />
-                                </a>
-                            )}
-                        </div>
-                    )}
+                            </div>
+                        ) : (
+                            <SidebarNavItem to="/organizations/about">About organizations</SidebarNavItem>
+                        ))}
```
Comment (valerybugakov) [inline client/web/src/user/settings/UserSettingsSidebar.tsx:94]: Let's remove the variable negation here and swap the left and right ternary expressions to make it easier to read: `!true ? 1 : 2 -> true ? 2 : 1`.

**Comment by rrhyne** [comment]
One last item... I improperly specified the properties on the metrics cards. We need to specify a key and value, where I supplied only a value. 

Here's an illustration of the problem and solution:

https://www.figma.com/file/IgD7ZzmNt4Un2nqkk1u1X4/Self-hosted-CTAs-in-the-Cloud-product?node-id=318%3A923

I've updated all the purple metrics cards to reflect this change.

Apologies!

File: client/web/src/org/OrgsArea.tsx (hunk 4 above)
Comment (gitstart-sourcegraph) [reply client/web/src/org/OrgsArea.tsx:58]: @valerybugakov  Could you please suggest a way to use `UserArea` with route `/organizations/about` (which route should be in `OrgArea`)? As far as I can see the `UserSettingsSideBar` requires user id from route

Comment (valerybugakov) [reply client/web/src/org/OrgsArea.tsx:58]: @gitstart-sourcegraph, you're right. The best option I see here is:
1. Move `AboutOrganizationPage` to `client/web/src/user/settings/aboutOrganization`
2. Move `AboutOrganizationPage` page route to `client/web/src/user/settings/routes.tsx`
3. Change the URL of the page from `/organizations/about` to `/users/admin/settings/about-organizations`

That way, we will be able to keep `UserSettingsSideBar` on the page. Let me know I miss something in this suggestion.

**Review by rrhyne — APPROVED** [review]
I ran through the pages, rendering looks really good on small sizes, link targets are correct and thanks for making the change to the tracking. 

Lets merge ASAP!!! ;)

**Pull request merged**